lib/ruby_indexer/lib/ruby_indexer/index_rbs.rb

```diff
@@ -0,0 +1,45 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class IndexRbs
+    extend T::Sig
+
+    sig { params(index: Index).void }
+    def initialize(index)
+      @index = index
+    end
+
+    sig { void }
+    def index_core_classes
+      loader = RBS::EnvironmentLoader.new
+      RBS::Environment.from_loader(loader).resolve_type_names
+
+      loader.each_signature do |source, pathname, _buffer, declarations, _directives|
+        process_signature(source, pathname, declarations)
+      end
+    end
+
+    private
+
+    sig { params(source: T.untyped, pathname: Pathname, declarations: T::Array[RBS::AST::Declarations::Base]).void }
+    def process_signature(source, pathname, declarations)
+      declarations.each do |declaration|
+        process_declaration(declaration, pathname)
+      end
+    end
+
+    sig { params(declaration: RBS::AST::Declarations::Base, pathname: Pathname).void }
+    def process_declaration(declaration, pathname)
+      return unless declaration.is_a?(RBS::AST::Declarations::Class)
+
+      nesting = [declaration.name.name.to_s]
+      file_path = pathname.to_s
+      location = RubyIndexer::Location.from_rbs_location(declaration.location)
+      comments = Array(declaration.comment&.string&.lines)
+      parent_class = nil
+      class_entry = Entry::Class.new(nesting, file_path, location, comments, parent_class)
+      @index << class_entry
+    end
```

Comment:
Since we're going to process multiple declaration types coming from RBS, can we already setup a case statement that only processes classes?

For example

```ruby
declarations.each do |declaration|
  case declaration
  when RBS::AST::Declarations::Class
    handle_class(declaration)
  else
    # Unhandled RBS declaration. Do nothing
  end
end
```